haskellPackages: update hackage and stackage; GHC 9.2.4

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #183112. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.compiler: ghc923 -> ghc924

https://www.haskell.org/ghc/download_ghc_9_2_4.html

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
all-cabal-hashes: 2022-07-27T11:03:11Z -> 2022-07-31T09:12:37Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: stackage LTS 19.16 -> LTS 19.17

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskellPackages: reflect ghc-lib*_9_2* updates

sternenseemann committed:
haskellPackages.uniform-error: update override comment

Seemingly the package is not yet feature complete and a desired feature
is already part of the test suite?!

sternenseemann committed:
haskellPackages.levenshtein: remove broken flag

Compile failure has been resolved upstream!

Closes #183151.

sternenseemann committed:
haskellPackages.cabal-hoogle: disable cabal v2 layout dependent test

sternenseemann committed:
haskellPackages.debian: drop obsolete patches

debian-4.0.4 has added GHC 9.0 support in a released version.

sternenseemann commented:
Only blocker I'm aware of is https://github.com/wz1000/HieDb/issues/46 which breaks HLS. Not sure how bad this issue is nor how it impacts haskell-language-server exactly.

github-actions[bot] committed:
Merge master into haskell-updates

maralorn commented:
I guess the worst case is having less precise type signatures displayed at some points? Don’t think that would be terrible.

maralorn committed:
haskell.packages.ghc924.hiedb: dontCheck

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

sternenseemann committed:
haskellPackages.uniform-fileio: disable tests incompatbile with HTF >= 0.15

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
maintainers/scripts/haskell/hydra-report: fix capitalisation

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
haskell.compiler.ghc941: apply fix for racy build system

https://gitlab.haskell.org/ghc/ghc/-/merge_requests/8751
https://gitlab.haskell.org/ghc/ghc/-/issues/21934

Hopefully this patch fixes the problem we experienced on Hydra once (out
of three builds).

mlen commented:
Is there any chance to backport `[haskell.packages.ghc924.hiedb: dontCheck](https://github.com/NixOS/nixpkgs/pull/184194/commits/3ca851a1d17940d1bb25e73784bfc1ea8d129779)` to 22.05? This seems to affect stable too: `nix-shell -p haskell-language-server` fails because `hiedb` tests fail.

cdepillabout commented:
@mlen We generally accept PRs to 22.05 for Haskell related stuff, especially if they get more stuff working.  Would you be able to open a PR for this?

caryoscelus commented:
@mlen @cdepillabout i've made a PR https://github.com/NixOS/nixpkgs/pull/189082/ , hopefully it's enough to fix build issue
